setuptools/dist.py

```diff
@@ -144,12 +144,14 @@ def read_pkg_file(self, file):
     self.license_files = _read_list_from_msg(msg, 'license-file')
 
 
-def single_line(val):
-    # quick and dirty validation for description pypa/setuptools#1390
+def ensure_summary_single_line(val):
+    """Validate that the summary does not have line breaks."""
```

Comment:
I try to avoid over-specifying the behavior. The original name of this function would validate to `single_line` for any field. I'd like to keep it decoupled from the context, and instead let the context of the traceback (`single_line(self.get_description())`) guide the user to the cause.

If we were to generalize the approach indicated by the current proposal, it would imply that `str(version)` should also be replaced by `ensure_version_str`, with its own docstring and customized error message.

Although it would be nice to provide more contextual error messages, the advantages of composition outweigh the disadvantages of requiring a bespoke validator for each field. If we really care about contextual error messages, we should refactor to model validators and fields and use that to provide the user more context about which fields failed validation.

Reply:
I chose to name it this way exactly because of the misleading purpose: the name suggested that it's universal but it was only used for the summary line + the error was specific to it too. I figured that if there's a desire to make it truly universal, it's out of the scope and should be done separately. Meanwhile, I've made the name correspond to what it actually does.

Reply:
I've changed it back in a4b7caeaa. I'd like to keep the code as simple and uncomplicated as possible.